src/LaunchDarkly.EventSource/EventSourceService.cs

```diff
@@ -92,7 +92,7 @@ public EventSourceService(Configuration configuration)
                     {
                         using (var reader = new StreamReader(stream))
                         {
-                            while (!cancellationToken.IsCancellationRequested && !reader.EndOfStream)
+                            while (!cancellationToken.IsCancellationRequested)
```

Comment:
Why don't we want `!reader.EndOfStream`?

Reply:
Strangely, accessing `reader.EndOfStream` causes `readTimeoutTask` to never resolve. Even a simple logging statement which accesses `reader.EndOfStream` reproduces this behavior.